src/troubleshooting.rst

```diff
@@ -0,0 +1,482 @@
+activate command.
+
+Sadly, some of these scripts don't defend against command failure or unset
+environment variables causing them to fail when configured in Cylc ``*script``
+(see also :ref:`troubleshooting.my_job_failed` for details).
+
+To avoid this, run ``set +eu`` before activating your environment. This turns
+off some Bash safety features allowing environment activation to complete.
+Remember to run ``set -eu`` afterwards to turn these features back on.
+
+.. code-block:: cylc
+
+   [runtime]
+       [[my_task]]
+            script = """
+               set +eu
+               conda activate <my_environment>
+               set -eu
+
+               do-something
+               do-something-else
+            """
+
+
+Error Messages
+--------------
+
+Cylc should normally present you with a simple, short error message when things
+go wrong.
+
+To see the full traceback, run the command / workflow in debug mode, e.g. using
+the ``--debug`` option.
+
+If you are presented with traceback when you are *not* running in debug mode,
+then this is not an expected error, please report the traceback to us.
+
+
+``FileNotFoundError: No such file or directory``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+This is the error message Python gives when you try to call an executable which
+does not exist in the ``$PATH``. It means there's something wrong with the Cylc
+installation.
+
+E.G. the following error:
+
+.. code-block::
+
+   FileNotFoundError: [Errno 2] No such file or directory: 'ssh'
+
+Means that ``ssh`` is not installed.
+
+See :ref:`non-python-requirements` for details on system requirements.
+
+
+.. _troubleshooting.remote_init_did_not_complete:
+
+``platform: <name> - initialisation did not complete``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+This means that Cylc was unable to install the required workflow files onto
+a remote platform.
+
+This either means that:
+
+1. The platform is down (e.g. all login nodes are offline).
+2. There is a network problem (e.g. you cannot connect to the login nodes).
+3. The platform is not correctly configured.
+
+Check the scheduler log, you might find some stderr associated with this
+message.
+
+If your site has configured this platform for you, it's probably (1) or (2),
+check you are able to access the platform and notify the site administrator as
+appropriate.
+
+If you are in the progress of setting up a new platform, it's probably (3).
+You might want to check that you've configured the
+:cylc:conf:`global.cylc[platforms][<platform name>]install target` correctly,
+note that this defaults to the platform name if not explicitly set.
+
+
+``OperationalError: disk I/O error``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+This means that something was unable to write to a file when it would expect to
+have been able to.
+
+This error usually occurs if when you have exceeded you filesystem quota.
+
+If a Cylc workflow cannot write to the filesystem, it will shutdown. Once
```

Comment:
```suggestion
If a Cylc scheduler cannot write to the filesystem, it will shutdown. Once
```
This has wider implications, so could leave for later. We've all long been using "scheduler" and "workflow" (meaning "running workflow") interchangeably, but I'd prefer not to conflate the two fundamentally different concepts.